tests/parser/features/decorators/test_private.py

```diff
@@ -12,4 +12,387 @@ def returnten() -> int128:
+    return a + b + c
+    """
+
+    c = get_contract_with_gas_estimation(private_test_code)
+    assert c.return_it() == 555
+
+
+def test_private_with_args_nested(get_contract_with_gas_estimation):
+    private_test_code = """
+@private
+def multiply(a: uint256, b: uint256) -> uint256:
+    c: uint256 = 7
+    d: uint256 = 8
+    e: uint256 = 9
+    return a * b
+
+@private
+def add_times2(a: uint256, b: uint256) -> uint256:
+    return self.multiply(3, (a + b))
+
+@public
+def return_it() -> uint256:
+    a: uint256 = 111
+    b: uint256 = 222
+    c: uint256 = self.add_times2(0, a)
+    return a + b + c
+    """
+
+    c = get_contract_with_gas_estimation(private_test_code)
+    assert c.return_it() == 666
+
+
+def test_private_bytes(get_contract_with_gas_estimation):
+    private_test_code = """
+greeting: public(bytes[100])
+
+@public
+def __init__():
+    self.greeting = "Hello "
+
+@private
+def construct(greet: bytes[100]) -> bytes[200]:
+    return concat(self.greeting, greet)
+
+@public
+def hithere(name: bytes[100]) -> bytes[200]:
+    d: bytes[200] = self.construct(name)
+    return d
+    """
+
+    c = get_contract_with_gas_estimation(private_test_code)
+    assert c.hithere(b"bob") == b"Hello bob"
+    assert c.hithere(b"alice") == b"Hello alice"
+
+
+def test_private_statement(get_contract_with_gas_estimation):
+    private_test_code = """
+greeting: public(bytes[20])
+
+@public
+def __init__():
+    self.greeting = "Hello "
+
+@private
+def set_greeting(_greeting: bytes[20]):
+    a: uint256 = 333
+    b: uint256 = 334
+    c: uint256 = 335
+    d: uint256 = 336
+    if a + b + c + d == 1338:
+        self.greeting = _greeting
+
+@public
+def construct(greet: bytes[20]) -> bytes[40]:
+    return concat(self.greeting, greet)
+
+@public
+def iprefer(_greeting: bytes[20]):
+    a: uint256 = 112
+    b: uint256 = 211
+    self.set_greeting(_greeting)
+    assert a == 112
+    assert b == 211
+
+@public
+def hithere(name: bytes[20]) -> bytes[40]:
+    d: bytes[40] = self.construct(name)
+    return d
+    """
+
+    c = get_contract_with_gas_estimation(private_test_code)
+    assert c.greeting() == b"Hello "
+    assert c.hithere(b"Bob") == b"Hello Bob"
+    c.iprefer(b'Hi there, ', transact={})
+    assert c.hithere(b"Alice") == b"Hi there, Alice"
+
+
+def test_private_default_parameter(get_contract_with_gas_estimation):
+    private_test_code = """
+@private
+def addition(a: uint256, b: uint256 = 1) -> uint256:
+    return a + b
+
+
+@public
+def add_one(a: uint256) -> uint256:
+    return self.addition(a)
+
+@public
+def added(a: uint256, b: uint256) -> uint256:
+    c: int128 = 333
+    d: uint256 = self.addition(a, b)
+    assert c == 333
+
+    return d
+    """
+
+    c = get_contract_with_gas_estimation(private_test_code)
+
+    assert c.add_one(20) == 21
+    assert c.added(10, 20) == 30
+
+
+def test_private_return_tuple(get_contract_with_gas_estimation):
+    code = """
+@private
+# @public
```

Comment:
Remove this comment? Is this comment intentional?

If intentional, it should be added to syntax tests not here.